### PR TITLE
refactor: persist subroutes, upgrade GoRouter

### PR DESCRIPTION
**Pull request by nesquikm**
## Description

This PR provides several routing-related things such as:
* Upgrade GoRouter to 9.0.0
* Persist subroutes after root route switching
* Add isSaveSubroutes for root routes

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/app/router/app_route.dart [hunk 1]
```diff
@@ -193,7 +201,9 @@ extension NavigationHelper on BuildContext {
   void goFurther(String location, {Object? extra}) {
     if (!mounted) return;
 
-    var resultLocation = Uri.parse(GoRouter.of(this).location);
+    final currentLocation = inject<NavigationService>().location;
+    // final currentLocation1 = GoRouterState.of(this).location;
```
Comment (Alex-A4) [inline lib/app/router/app_route.dart:205]: Forgot removed comment

File: lib/app/router/router.dart [hunk 2]
```diff
@@ -14,11 +14,21 @@ export 'routs/routs.dart';
 
 // ignore: long-method
 GoRouter getRouter(BuildContext _) {
+  // Last saved root app route
+  AppRoute? lastRootAppRoute;
+
+  // Just for
```
Comment (Alex-A4) [inline lib/app/router/router.dart:20]: Not finished comment

File: lib/app/router/router.dart [hunk 3]
```diff
@@ -31,22 +41,79 @@ GoRouter getRouter(BuildContext _) {
     return null;
   }
 
+  // Set location, it will be called in multiple places
+  // because GoRouter's 'redirect' handler doesn't call it aster pop()
```
Comment (Alex-A4) [inline lib/app/router/router.dart:45]: `after` expect of `aster`

**Review by Alex-A4 — APPROVED** [review]

File: lib/app/router/router.dart (hunk 3 above)
Comment (nesquikm) [reply lib/app/router/router.dart:45]: 🤟

File: lib/app/router/router.dart (hunk 2 above)
Comment (nesquikm) [reply lib/app/router/router.dart:20]: 🤟

File: lib/app/router/app_route.dart (hunk 1 above)
Comment (nesquikm) [reply lib/app/router/app_route.dart]: 🤟

**Pull request merged**